geostore/check.py

```diff
@@ -5,8 +5,9 @@ class Check(Enum):
     ASSETS_IN_DATASET = "assets in dataset"
     CHECKSUM = "checksum"
     DUPLICATE_OBJECT_KEY = "duplicate asset name"
+    FILE_NOT_FOUND = "file not found in staging or storage"
```

Comment:
I just realised these errors should probably be fields on the relevant exception classes. For example, this one should be on the `DatasetFileNotFoundError` class. But that's something for another PR.

Reply:
Do you mean in addition to here also?

Most of these "Checks" don't correspond to a defined Exception.